scheduling/shortest_deadline_first.py

```diff
@@ -0,0 +1,70 @@
+"""
+Earliest Deadline First (EDF) Scheduling Algorithm
+
+This code implements the Earliest Deadline First (EDF)
+scheduling algorithm, which schedules processes based on their deadlines.
+If a process cannot meet its deadline, it is marked as "Idle."
+
+Reference:
+https://www.geeksforgeeks.org/
+earliest-deadline-first-edf-cpu-scheduling-algorithm/
+
+Author: Arunkumar
+Date: 14th October 2023
+"""
+
+
+def earliest_deadline_first_scheduling(
+    processes: list[tuple[str, int, int, int]]
+) -> list[str]:
+    """
+    Perform Earliest Deadline First (EDF) scheduling.
+
```

Comment:
Input Validation: You may want to add some input validation to ensure that the input list of processes is well-formed. For example, you can check if the arrival time is less than or equal to the deadline for each process

Reply:
Once the code is merged, make a pull request on enhancement and do it!

Reply:
If you could, make a review on this PR too.   #10432